src/client.rs

```diff
@@ -103,23 +117,293 @@ impl<'a> Client<'a> {
+            subject_key,
+            subject_attributes,
+            Some(VariationType::Integer),
+            |x| {
+                x.as_integer()
+                    // The unwrap cannot fail because the type is checked during evaluation.
+                    .unwrap()
+            },
+        )
+    }
+
+    /// Retrieves the assignment value for a given feature flag and subject as a numeric value.
+    ///
+    /// If the subject is not eligible for any allocation, returns `Ok(None)`.
+    ///
+    /// If the configuration has not been fetched yet, returns `Ok(None)`.
+    /// You should call [`Client::start_poller_thread`] before any call to
+    /// `get_numeric_assignment()`. Otherwise, the client will always return `None`.
+    ///
+    /// It is recommended to wait for the Eppo configuration to get fetched with
+    /// [`PollerThread::wait_for_configuration()`].
+    ///
+    /// # Errors
+    ///
+    /// Returns an error in the following cases:
+    /// - [`Error::FlagNotFound`] if the requested flag configuration was not found.
+    /// - [`Error::InvalidType`] if the requested flag has an invalid type or type conversion fails.
+    /// - [`Error::ConfigurationParseError`] or [`Error::ConfigurationError`] if the configuration
+    /// received from the server is invalid.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # fn test(client: &eppo::Client) {
+    /// let assignment = client
+    ///     .get_numeric_assignment("a-num-flag", "user-id", &[
+    ///         ("age".to_owned(), 42.0.into())
+    ///     ].iter().cloned().collect())
+    ///     .unwrap_or_default()
+    ///     .unwrap_or(0.0);
+    /// # }
+    /// ```
+    pub fn get_numeric_assignment(
+        &self,
+        flag_key: &str,
+        subject_key: &str,
+        subject_attributes: &SubjectAttributes,
+    ) -> Result<Option<f64>> {
+        self.get_assignment_inner(
+            flag_key,
+            subject_key,
+            subject_attributes,
+            Some(VariationType::Numeric),
+            |x| {
+                x.as_numeric()
+                    // The unwrap cannot fail because the type is checked during evaluation.
+                    .unwrap()
+            },
+        )
+    }
+
+    /// Retrieves the assignment value for a given feature flag and subject as a boolean value.
+    ///
+    /// If the subject is not eligible for any allocation, returns `Ok(None)`.
+    ///
+    /// If the configuration has not been fetched yet, returns `Ok(None)`.
+    /// You should call [`Client::start_poller_thread`] before any call to
+    /// `get_boolean_assignment()`. Otherwise, the client will always return `None`.
+    ///
+    /// It is recommended to wait for the Eppo configuration to get fetched with
+    /// [`PollerThread::wait_for_configuration()`].
+    ///
+    /// # Errors
+    ///
+    /// Returns an error in the following cases:
+    /// - [`Error::FlagNotFound`] if the requested flag configuration was not found.
```

Comment:
This could happen if someone adds the `get_*_assignment` function to their code but hasn't yet configured a flag in eppo.cloud. This is a normal flow and from the perspective of the client, should not throw an error.

What I want to happen, from the client's perspective, is the `default_value` be returned. However the `FlagNotFound` can be returned in some way to inform users of the reason for the value.